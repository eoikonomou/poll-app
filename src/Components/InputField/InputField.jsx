import React from 'react';
import PropTypes from 'prop-types';

const InputField = ({
                      value,
                      onChange,
                      id,
                      placeholder,
                      disabled,
                    }) => (
  <input
    id={id}
    className="poll-question"
    placeholder={placeholder}
    value={value}
    onChange={onChange}
    maxLength={80}
    disabled={disabled}
  />
);

InputField.displayName = 'InputField';

InputField.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  id: PropTypes.string,
  placeholder: PropTypes.string,
  disabled: PropTypes.bool,
};

InputField.defaultProps = {
  id: '',
  placeholder: 'Input the question here...',
  disabled: false,
};

export default InputField;
