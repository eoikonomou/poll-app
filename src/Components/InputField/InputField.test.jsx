import React from 'react';
import { shallow } from 'enzyme';
import InputField from './InputField';

test('InputField renders properly with the correct props', () => {
  const onChangeFunc = () => {};
  const wrapper = shallow(
    <InputField
      value="test-value"
      onChange={onChangeFunc}
      id="test-id"
      placeholder="test-placeholder"
      disabled
    />,
  );
  expect(wrapper.prop('value')).toEqual('test-value');
  expect(wrapper.prop('id')).toEqual('test-id');
  expect(wrapper.prop('placeholder')).toEqual('test-placeholder');
  expect(wrapper.prop('disabled')).toBe(true);
  expect(wrapper.prop('onChange')).toEqual(onChangeFunc);
  expect(wrapper).toMatchSnapshot();
});

test('InputField logs error if value or onChange properties are missing', () => {
  const spy = jest.spyOn(console, 'error');
  shallow(
    <InputField
      onChange={() => {}}
      id="test-id"
      placeholder="test-placeholder"
      disabled={false}
    />,
  );
  shallow(
    <InputField
      value=""
      id="test-id"
      placeholder="test-placeholder"
      disabled={false}
    />,
  );
  expect(spy).toHaveBeenCalledTimes(2);
});

test('InputField default properties are properly set', () => {
  const wrapper = shallow(
    <InputField
      value=""
      onChange={() => {}}
    />,
  );
  expect(wrapper.prop('id')).toEqual('');
  expect(wrapper.prop('placeholder')).toEqual('Input the question here...');
  expect(wrapper.prop('disabled')).toBe(false);
});
