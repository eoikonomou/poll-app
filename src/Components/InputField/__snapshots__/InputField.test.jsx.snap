// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField should render properly with the correct props 1`] = `
<input
  className="poll-question"
  disabled={true}
  id="test-id"
  maxLength={80}
  onChange={[Function]}
  placeholder="test-placeholder"
  value="test-value"
/>
`;
